.semgrep/adapter/type-bid-assignment.yml

```diff
@@ -2,8 +2,11 @@ rules:
   - id: type-bid-assignment-check
     languages:
       - go
-    message: >
-      Found incorrect assignment made to `Bid` Consider using `seatBids.Bid` loop index as shown below  
+    message: >      
+      Found incorrect assignment made to $KEY. $BID variable receives a new value in each iteration of range loop. Assigning the address of $BID `(&$BID)` to $KEY will result in a pointer that always points to the same memory address with the value of the last iteration. 
+      This can lead to unexpected behavior or incorrect results. Refer https://go.dev/play/p/9ZS1f-5h4qS
+
+      Consider using `seatBids.Bid` loop index as shown below  
```

Comment:
**Nitpick:** Consider rewriting the last sentence as:
`Consider using an index variable in the seatBids.Bid loop as shown below`

Reply:
https://github.com/prebid/prebid-server/pull/2833/commits/bd7aaffb9104aeda884a420f785bbdcef74786c3